### ISSUE TITLE
Add support for WDS Coding Standards

### ISSUE THREAD

**Issue by gregrickaby** [WebDevStudios/wd_s#548]
Make sure both @wordpress/scripts and code editors are able to leverage the WDS Coding Standards. This will require some help from @aubreypwd.

- [ ] Code editors should be able to find coding standard configs 
- [ ] Code editors should be able to auto-format on save (yes, this includes support for Prettier)
- [ ] @wordpress/scripts should also be linting 
- [ ] Possibly bring in [Husky](https://github.com/typicode/husky) and use [Lint Staged](https://github.com/okonet/lint-staged) to enforce coding standards on the `pre-commit` hook. (See Greg if you have questions)

**Comment by aubreypwd** [comment]
> Code editors should be able to find coding standard configs

If you install `*-coding-standards` per normal (the instructions on repos) this should be the case as long as editors are configured correctly. I do know that VSCode does not always behave properly unless you have the root of your editor where the `package.json`, `phpcs.xml` files are, and PHP Storm has to be configured per project.

I think the effort that has to be made here is making sure that no other lingering coding standard packages/things are not conflicting/getting in the way with our packages—as I have seen in the past.

I know `wd_s` is using a custom SASS/`stylelint` setup (the correct one, actually) and I will need time to move `css-coding-standards` along and finally integrate what `wds_s` is doing into the package officially.

> Code editors should be able to auto-format on save (yes, this includes support for Prettier)

I've talked with various people at our company about this and it's something I'm interested in getting caught up with at some point. I would recommend integrating this at the coding standards level (if possible) just so coding standards inherit the work for this.

I know @michealengland has said that he understand how to setup Prettier, so I think connecting with him is smart right now.

> @wordpress/scripts should also be linting

I have talked with @michealengland about helping me figure out if we need to get the linting you're mentioning here integrated with our current `js-coding-standard`'s.

> Possibly bring in Husky and use Lint Staged to enforce coding standards on the pre-commit hook.

I've made attempts at this in the past with little success so far. But I'm very interested to see this work so any NEW code written (e.g. in the _diff_, not the _entire_ codebase) is linted; this would be really cool to see!

---

So, if I had a wishlist for this, it would be for time:

- [ ] @aubreypwd and @coreymcollins time to work on `css-coding-standards` and get it caught up with `wd_s`, `stylelint`, etc
- [ ] @michealengland and @aubreypwd time to figure out how to improve `js-coding-standards` with Prettier
	- [ ] @aubreypwd time to integrate our Prettier solution into `wd_s`
- [ ] @michealengland and @aubreypwd time to review `@wordpress/scripts` and `js-coding-standards` and figure out what can be done to integrate it at the package level
- [x] @aubreypwd time to review `wd_s` coding standards, ensure there are no conflicts with other lingering things, and finalize
